include/private/tdymaterialpropertiesimpl.h

```diff
@@ -0,0 +1,26 @@
+#if !defined(TDYMATERIALPROPERTIES_H)
+#define TDYMATERIALPROPERTIES_H
+
+#include <petsc.h>
+
+typedef struct _MaterialProp MaterialProp;
```

Comment:
Instead of having a separate typedef, you could just include it in the struct's definition:

```
typedef struct {
    PetscReal *K, *K0;
    PetscReal *porosity;
    PetscReal *Kappa, *Kappa0;
    PetscReal *Cr;
    PetscReal *rhosoil;
} MaterialProp;
```

This results in only having one symbol related to `MaterialProp`, which is a basic struct not a "class"-style struct with a private definition. Not a big deal, but simpler.

Reply:
You can do the same thing for `CharacteristicCurve` as well. But like I said, it's not a big deal, so I'm fine with them as is.

Reply:
I will make suggested changes.